lib/inferno/dsl/runnable.rb

```diff
@@ -464,6 +464,21 @@ def required_suite_options(suite_option_requirements)
           end
       end
 
+      # Replace a child test/group
+      #
+      # @param id_to_replace (Symbol, String) The ID of the child to be replaced
+      # @param new_child (Inferno::TestGroup, Inferno::Test) The new child to replace the existing one
+      # @example
+      #   test from: :test1
+      #   test from: :test2
+      #   test from: :test3
+      #
+      #   replace :test2, new_test
+      def replace(id_to_replace, new_child)
```

Comment:
I think both arguments should be ids. If you have direct access to the test/group, it's easy to get the id.

I also think this needs to take a block which will be executed in the context of the new child to make it easier to configure:

```ruby
replace(:some_id, :some_other_id) do
  id :new_id
  config(...)
end
```